pkg/languages/js/js.go

```diff
@@ -0,0 +1,171 @@
+package js
+
+import (
+	"context"
+	"fmt"
+	"github.com/containers/image/types"
+	"github.com/sirupsen/logrus"
+	"github.com/slinkydeveloper/kfn/pkg/config"
+	"github.com/slinkydeveloper/kfn/pkg/image"
+	"github.com/slinkydeveloper/kfn/pkg/languages"
+	"github.com/slinkydeveloper/kfn/pkg/util"
+	"io/ioutil"
+	"os"
+	"path"
+)
+
+const (
+	baseImage          = "node:12-alpine"
+	jsRuntimeRemoteZip = "https://github.com/openshift-cloud-functions/faas-js-runtime-image/archive/master.zip"
+)
+
+type jsLanguageManager struct {
+	resourceLoader util.ResourceLoader
+}
+
+func NewJsLanguageManger() languages.LanguageManager {
+	return jsLanguageManager{util.NewResourceLoader("../../bootstrap_templates/js")}
+}
+
+func (r jsLanguageManager) Bootstrap(functionName string, targetDirectory string) error {
+	err := util.MkdirpIfNotExists(targetDirectory)
+	if err != nil {
+		return err
+	}
+
+	main, err := r.resourceLoader.LoadResource("index.js")
+	if err != nil {
+		return err
+	}
+
+	packageJsonTempl, err := r.resourceLoader.LoadTemplate("package.json")
+	if err != nil {
+		return err
+	}
+
+	err = util.PipeTemplateToFile(path.Join(targetDirectory, "package.json"), packageJsonTempl, struct {
+		FunctionName string
+	}{functionName})
+	if err != nil {
+		return err
+	}
+
+	return util.WriteFiles(
+		targetDirectory,
+		util.WriteDest{fmt.Sprintf("%s.js", functionName), main},
+	)
+}
+
+// For JS Runtime nothing is required, since the building atm is done directly in the container
+func (j jsLanguageManager) CheckCompileDependencies() error {
+	return nil
+}
+
+func (j jsLanguageManager) Compile(inputFile string) (string, []string, error) {
```

Comment:
Does this basically just do nothing because the build happens on the image?

Reply:
in JS case yes, but in rust for example it will invoke `cargo build`